.github/workflows/pages-deployment.yml

```diff
@@ -25,6 +25,8 @@ jobs:
           python3 -m pip install -r ./requirements.txt
 
       - name: Build site
+        env:
+          AccessToken: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
question: Could we limit this token's permission? For example:

```yaml
permissions:
  contents: read

jobs:
  …
```

***

I also note that Cloudflare's publish action [uses it for deployments](https://github.com/cloudflare/pages-action/blob/1aff065c9e5139731bb0e61b9796b800c22279a0/README.md#usage). So actually the configuration might be something like:

```yaml
permissions:
  contents: read
  deployments: write
```

***

That said, I'm not entirely sure why the token is needed at all for the multirepo plugin. The documentation for the multirepo plugin does [state that you must provide a token](https://github.com/jdoiro3/mkdocs-multirepo-plugin/blob/4f7820717fcf687576075ec14813ecc71de06535/README.md#github-actions). But the actual cloning script used by it seems — to my naive eyes — to fallback to attempting to [clone without a token, if none is provided](https://github.com/jdoiro3/mkdocs-multirepo-plugin/blob/4f7820717fcf687576075ec14813ecc71de06535/mkdocs_multirepo_plugin/scripts/sparse_clone.sh#L12-L19).

Then again, without a token you may encounter rate limits to the IP address in GitHub Actions runners, I think, which could cause random failures. So that might be an argument for using a token with limited permissions.

Reply:
It does need the token - the repo doesn't get cloned without it (I tried).  I think it may be specific to cloning in Github Actions - it doesn't seem to need a token or auth locally.
We can try limiting the token; do we do that in other repos (and should we)?

Reply:
Oh, OK! That's strange, because it looks like from the code that it should work, but I only glanced at it. Maybe it's different in GitHub Actions specifically to avoid rate-limiting failures.

> We can try limiting the token; do we do that in other repos (and should we)?

To my knowledge, we don't have any organisational policy on this. So maybe that's worth asking elsewhere.

(My personal view is: yes, based on the principle of least privilege. I think it's generally worth considering what the GitHub Actions tokens can do, particularly when interacting with third-party actions. The defaults are [quite permissive](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), unless they are already locked down by the organisation owners.)

Reply:
Done and all looks fine in the preview